inference/core/workflows/core_steps/transformations/size_measurement/v1.py

```diff
@@ -0,0 +1,171 @@
+        kind=[
+            INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            OBJECT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(
+        description="Predictions from the reference object model",
+        examples=["$segmentation.reference_predictions"],
+    )
+    object_predictions: StepOutputSelector(
+        kind=[
+            INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            OBJECT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(
+        description="Predictions from the model that detects the object to measure",
+        examples=["$segmentation.object_predictions"],
+    )
+    reference_dimensions: Union[str, Tuple[float, float]] = Field(
+        description="Dimensions of the reference object (width, height) in desired units (e.g., inches) as a string in the format 'width,height' or as a tuple (width, height)",
+        examples=["5.0,5.0", (5.0, 5.0)],
+    )
+
+    @validator("reference_dimensions", pre=True)
+    def parse_reference_dimensions(
+        cls, value: Union[str, Tuple[float, float]]
+    ) -> Tuple[float, float]:
+        if isinstance(value, str):
+            try:
+                width, height = map(float, value.split(","))
+                return width, height
+            except ValueError:
+                raise ValueError(
+                    "reference_dimensions must be a string in the format 'width,height'"
+                )
+        elif isinstance(value, tuple) and len(value) == 2:
+            return value
+        else:
+            raise ValueError(
+                "reference_dimensions must be a string in the format 'width,height' or a tuple (width, height)"
+            )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name=OUTPUT_KEY, kind=[LIST_OF_VALUES_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+def get_detection_dimensions(
+    detection: sv.Detections, index: int
+) -> Tuple[float, float]:
+    if detection.mask is not None:
+        mask = detection.mask[index].astype(np.uint8)
+        x, y, w, h = cv.boundingRect(mask)
+    else:
+        bbox = detection.xyxy[index]
+        x, y, w, h = bbox[0], bbox[1], bbox[2] - bbox[0], bbox[3] - bbox[1]
+    return w, h
+
+
+class SizeMeasurementBlockV1(WorkflowBlock):
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return SizeMeasurementManifest
+
+    def run(
+        self,
+        reference_predictions: Batch[sv.Detections],
+        object_predictions: Batch[sv.Detections],
+        reference_dimensions: Union[str, Tuple[float, float]],
+    ) -> BlockResult:
+        result = []
+        ref_width_actual, ref_height_actual = (
+            SizeMeasurementManifest.parse_reference_dimensions(reference_dimensions)
```

Comment:
If I understand correctly,  `@validator("reference_dimensions", pre=True)` should make the transformation while manifest parsing and you would only be able to get `Tuple[float, float]` as `reference_dimensions`, but I may be mistaken